src/lambda.ts

```diff
@@ -0,0 +1,134 @@
+import type {
+  APIGatewayEvent,
+  APIGatewayProxyEventV2,
+  APIGatewayProxyResult,
+  Context,
+} from "aws-lambda";
+import {
+  InngestCommHandler,
+  ServeHandler,
+} from "./components/InngestCommHandler";
+import { headerKeys, queryKeys } from "./helpers/consts";
+import { allProcessEnv } from "./helpers/env";
+
+/**
+ * With AWS Lambda, serve and register any declared functions with Inngest,
+ * making them available to be triggered by events.
+ *
+ * @example
+ *
+ * ```ts
+ * import { Inngest } from "inngest";
+ * import { serve } from "inngest/lambda";
+ *
+ * const inngest = new Inngest({ name: "My Lambda App" });
+ *
+ * const fn = inngest.createFunction(
+ *   { name: "Hello World" },
+ *   { event: "test/hello.world" },
+ *   async ({ event }) => {
+ *     return "Hello World";
+ *   }
+ * );
+ *
+ * export const handler = serve(inngest, [fn]);
+ * ```
+ *
+ * @public
+ */
+export const serve: ServeHandler = (nameOrInngest, fns, opts) => {
+  const handler = new InngestCommHandler(
+    "aws-lambda",
+    nameOrInngest,
+    fns,
+    { ...opts },
+    (event: APIGatewayEvent | APIGatewayProxyEventV2, _context: Context) => {
+      /**
+       * Try to handle multiple incoming event types, as Lambda can have many
+       * triggers.
+       *
+       * This still doesn't handle all cases, but it's a start.
+       */
+      const eventIsV2 = ((
+        ev: APIGatewayEvent | APIGatewayProxyEventV2
+      ): ev is APIGatewayProxyEventV2 => {
+        return (ev as APIGatewayProxyEventV2).version === "2.0";
+      })(event);
+
+      const method = eventIsV2
+        ? event.requestContext.http.method
+        : event.httpMethod;
+      const path = eventIsV2 ? event.requestContext.http.path : event.path;
+
+      const env = allProcessEnv();
+      const proto = event.headers["x-forwarded-proto"] || "https";
+
+      const url = new URL(path, `${proto}://${event.headers.host || ""}`);
+
+      const isProduction =
+        env.CONTEXT === "production" ||
+        env.ENVIRONMENT === "production" ||
+        env.NODE_ENV?.startsWith("prod") ||
+        false;
+
+      return {
+        register: () => {
+          if (method === "PUT") {
```

Comment:
I'm curious, why does every handler need to have these conditional statements within each handler function? Could we instead pass the `method` as a string to the return statement like `register` and `run`? it feels like it's duplicative in these handlers.  i.e.

```ts
// for this example
return {
  method,
  register : () => {...
}
// for other handlers that use "req"
return {
  method: req.method,
  register : () => {...
}
```